json-decode.go

```diff
@@ -32,14 +31,15 @@ func (cdc *Codec) decodeReflectJSON(bz []byte, info *TypeInfo, rv reflect.Value,
 
 	// Read disfix bytes if registered.
 	if info.Registered {
-		// Strip the disfix bytes after checking it.
-		var disfix DisfixBytes
-		disfix, bz, err = decodeDisfixJSON(bz)
+		// Strip the wrapper after checking it.
+		var name string
+		name, bz, err = decodeInterfaceJSON(bz)
 		if err != nil {
 			return
 		}
-		if !info.GetDisfix().EqualBytes(disfix[:]) {
-			err = fmt.Errorf("Expected disfix bytes %X but got %X", info.GetDisfix(), disfix)
+		if info.Name != name {
+			err = fmt.Errorf("Expected type name %s but got %s",
```

Comment:
minor nitpick: error strings should not be capitalized. See https://github.com/golang/go/wiki/CodeReviewComments#error-strings

Reply:
Seems like English grammar is bleeding into code architecture which seems backwards to me, but ok sure.